bin/steps/python

```diff
@@ -24,37 +26,35 @@ if [ ! "$SKIP_INSTALL" ]; then
     # Prepare destination directory.
     mkdir -p .heroku/python
 
-    curl https://lang-python.s3.amazonaws.com/$STACK/runtimes/$PYTHON_VERSION.tar.gz -s | tar zxv -C .heroku/python &> /dev/null
     mcount "version.python.$PYTHON_VERSION"
 
-    if [[ $? != 0 ]] ; then
+    if ! curl "https://lang-python.s3.amazonaws.com/$STACK/runtimes/$PYTHON_VERSION.tar.gz" -s | tar zxv -C .heroku/python &> /dev/null; then
       puts-warn "Requested runtime ($PYTHON_VERSION) is not available for this stack ($STACK)."
       puts-warn "Aborting.  More info: https://devcenter.heroku.com/articles/python-support"
       exit 1
     fi
 
   # Record for future reference.
-  echo $PYTHON_VERSION > .heroku/python-version
-  echo $STACK > .heroku/python-stack
+  echo "$PYTHON_VERSION" > .heroku/python-version
+  echo "$STACK" > .heroku/python-stack
   FRESH_PYTHON=true
 
   hash -r
 fi
 
 # If Pip isn't up to date:
-if [ "$FRESH_PYTHON" ] || [[ ! $(pip --version) == *$PIP_VERSION* ]]; then
-  WORKING_DIR=$(pwd)
+if [ "$FRESH_PYTHON" ] || [[ ! $(pip --version) == *$PIP_UPDATE* ]]; then
 
   TMPTARDIR=$(mktemp -d)
-  trap "rm -rf $TMPTARDIR" RETURN
+  trap 'rm -rf $TMPTARDIR' RETURN
```

Comment:
I don't think the `TMPTARDIR` directory is actually used any more? (Was previously used for the manual pip installation before the switch to get-pip.py)